src/SSHDebugPS/AD7UnixAsyncCommand.cs

```diff
@@ -143,9 +143,25 @@ private void Close()
                     return;
 
                 _isClosed = true;
-            }
 
-            _streamingShell.Close();
+                try
+                {
+                    if (_streamingShell != null && _streamingShell.IsOpen)
+                    {
+                        _streamingShell.Close();
+                    }
+
+                    if (_streamingShell != null)
+                    {
+                        _streamingShell.Dispose();
+                        _streamingShell = null;
+                    }
+                }
+                catch (ThreadInterruptedException)
+                {
+                    // This can happen if the command failed on a timeout. Say dotnet restore failed because of insufficient permissions etc.
```

Comment:
This comment can be improved, it doesn't currently indicate that it's okay that we don't try to dispose again.